processor/src/operations/decorators.rs

```diff
@@ -7,9 +11,79 @@ impl Process {
     // DEBUGGING
     // --------------------------------------------------------------------------------------------
 
-    /// TODO: implement
-    pub fn op_debug(&self) -> Result<(), ExecutionError> {
-        unimplemented!();
+    /// Print out debugging information based on options passed
+    pub fn op_debug(&mut self, options: DebugOptions) -> Result<(), ExecutionError> {
+        match options {
+            DebugOptions::All => {
+                self.print_stack(None);
+                self.print_mem(None, None);
+            }
+            DebugOptions::Stack(n) => self.print_stack(n),
+            DebugOptions::Mem(n, m) => self.print_mem(n, m),
+            DebugOptions::Local() => self.print_local(),
+        }
+
+        Ok(())
+    }
+
+    // DEBUG HELPER
+    // ---------------------------------------------------------------------------------
+
+    /// Print stack information for debugging.
+    /// If n is passed, this prints the last n items in the stack.
```

Comment:
nit: would use "Prints" instead of "print". Also, would say "the top n" items (rather than "the last n items").